### PR TITLE
[MEASURES FORM API] Query improvements + RSPEC request tests

### PR DESCRIPTION
**Pull request by rusllonrails**
[TRELLO CARD](https://trello.com/c/fB2ulVCR/151-refactor-code-and-write-automated-tests-18)

### CODE REVIEW

File: app/models/certificate.rb [hunk 1]
```diff
@@ -15,6 +15,34 @@ class Certificate < Sequel::Model
...
+
+      if filter_ops[:q].present?
+        q_rule = "#{filter_ops[:q]}%"
+
+        scope = scope.join_table(:inner,
+          :certificate_descriptions,
+          certificate_type_code: :certificate_type_code,
+          certificate_code: :certificate_code
+        ).where("
+          certificates.certificate_code ilike ? OR
```
Comment (matthewford) [inline app/models/certificate.rb]: could be written in sequel SQL::Expression's

http://sequel.jeremyevans.net/rdoc/files/doc/querying_rdoc.html#label-SQL-3A-3AExpression

With string concat https://sequel.jeremyevans.net/rdoc/files/doc/dataset_filtering_rdoc.html#label-String+concatenation

File: app/models/chapter_section.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+class ChapterSection < Sequel::Model(:chapters_sections)
```
Comment (matthewford) [inline app/models/chapter_section.rb]: do we need this class?

Comment (rusllonrails) [reply app/models/chapter_section.rb]: @matthewford yep I'm using it in Rspec tests for linking Sections and Chapters.

```
  let!(:chapter) do
    ch = create(:chapter,
      goods_nomenclature_item_id: chapter_code,
      producline_suffix: "80",
      validity_start_date: Date.today
    )

    period = add_period(ch.goods_nomenclature_sid, chapter_code)
    add_description(period,
      ch.goods_nomenclature_sid,
      chapter_code,
      "Inorganic chemicals; organic or inorganic compounds of precious metals, of rare-earth metals, of radioactive elements or of isotopes"
    )

    create(:chapter_section,
      goods_nomenclature_sid: ch.goods_nomenclature_sid,
      section_id: section.id
    )

    ch
  end
```

Comment (matthewford) [reply app/models/chapter_section.rb]: ok

File: app/models/certificate.rb (hunk 1 above)
Comment (rusllonrails) [reply app/models/certificate.rb]: @matthewford agree will rewrite

**Pull request merged**